### PR TITLE
Fix/9904 wrong error message

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
FileScanner was broken after last refactoring - solved that by refactoring coordinator and viewModel
Added and alert for the case that certificate already is registered in the app.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9904

## Screenshots
Screen Shot 2021-10-07 at 23 21 06

### CODE REVIEW

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -3230,6 +3230,10 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
 
 "FileScanner_PasswordError_title" = "Falsches Passwort";
 
+"FileScanner_AlreadyRegistered_title" = "Signatur ungültig";
```
Comment (KathrinLuecke) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:3233]: The label of the error message should be: "Zertifikat schon registriert"

**Review by KathrinLuecke — CHANGES_REQUESTED** [review]
Please change label of error message.

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: I changed it. Please review again @KathrinLuecke

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 1 above)
[the remark this answers is not in the document]
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: Please have another look, @Ein-Tim :)

**Review by KathrinLuecke — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift [hunk 2]
```diff
@@ -220,6 +292,9 @@ class FileScannerCoordinator {
 	}
 
 	private func hideIndicator() {
+		guard activityIndicatorView.alpha >= 0.0 else {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift:295]: Has this guard any function? Can the can the value be '< 0' ?

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift [hunk 3]
```diff
@@ -4,64 +4,136 @@
...
 		self.qrCodeFound = qrCodeFound
 		self.noQRCodeFound = noQRCodeFound
 	}
 
+	// MARK: - UIImagePickerControllerDelegate
+
+	func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+		finishedPickingImage()
+
+		DispatchQueue.global(qos: .background).async { [weak self] in
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift:30]:
```suggestion
		DispatchQueue.global(qos: . userInitiated).async { [weak self] in
```
Seems better fitting, don't you think?
https://developer.apple.com/documentation/dispatch/dispatchqos

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift [hunk 4]
```diff
@@ -3,50 +3,20 @@
...
-			viewControllerSpy,
-			fileScannerViewModel: viewModel,
-			qrCodeFound: { _ in },
-			noQRCodeFound: { }
-		)
-		fileScannerCoordinator.start()
-
-		viewModel.processingStarted?()
-
-		XCTAssertTrue(viewControllerSpy.viewSpy.addSubViewCalled)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift:39]: ViewSpy can be removed. Its not used anymore as far as I see.

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift [hunk 5]
```diff
@@ -138,6 +108,19 @@ private class ViewControllerSpy: UIViewController {
...
+
+	func scan(_ image: UIImage) {}
+
+	func scan(_ pdfDocument: PDFDocument) {}
+
+	func unlockAndScan(_ pdfDocument: PDFDocument) {}
+
+	func processItemProvider(_ itemProvider: NSItemProvider) {}
+
 	var finishedPickingImage: (() -> Void)?
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift:124]: Can be removed.

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift (hunk 3 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinator.swift:30]: This is new, thanks for sharing @marcussc

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift (hunk 4 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift:39]: Still needed for other tests

File: src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift (hunk 5 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/FileScanner/__tests__/FileScannerCoordinatorTests.swift:124]: Yes!

**Pull request merged**